drivers/gpu/drm/vc4/vc4_kms.c

```diff
@@ -395,7 +395,7 @@ static void vc4_atomic_commit_tail(struct drm_atomic_state *state)
 		old_hvs_state->fifo_state[channel].pending_commit = NULL;
 	}
 
-	if (vc4->hvs && vc4->hvs->hvs5) {
+	if (vc4->is_vc5) {
```

Comment:
This breaks fkms (where `vc4->hvs` would be null). Suggest `if (vc4->is_vc5 && !vc4->firmware_kms)` as fix.